server/src/models/auth.model.ts

```diff
@@ -0,0 +1,14 @@
+import { Schema, model, models } from 'mongoose';
+import { ResetPwdTokenSchema } from '../types/auth.types';
+
+const resetPwdTokenSchema = new Schema<ResetPwdTokenSchema>({
+  email: { type: String, required: true },
+  token: { type: String, required: true },
+  expiresAt: { type: Number, required: true },
+  isUsed: { type: Boolean, default: false },
```

Comment:
if the token is used or expired, then you can just delete the document after. then you don't need to have the isUsed flag. once it's used, delete it. once it expired, delete it.

https://stackoverflow.com/questions/66885448/how-to-automatically-delete-document-in-mongodb-using-mongoose-on-some-date

mongoose have this expiresAt field